docs/DEVICE-CONNECTIVITY.md

```diff
@@ -1,142 +1,483 @@
+
+![vlans-and-lags](./images/eve-vlans-and-lags.png)
+
+### SystemAdapter
+
+SystemAdapter (term used in EVE API) assigns network config (DHCP, IP, DNS and NTP config)
+together with some logical attributes such as "usage" or [cost](#load-spreading-and-failover)
+to a physical network port or to a logical lower-layer network adapter (VLAN, LAG).
+
+SystemAdapter can be used to elevate physical/L2-only adapter to an IP-ready network endpoint.
+SystemAdapter may either instruct EVE to run DHCP client for the network interface or to assign
+IP config statically. However, even L2-only interface (to be used with switch NI), still needs
+SystemAdapter attached for it to be referenceable by higher-level objects like network instances.
+
+SystemAdapter determines the usage of the network port, which is one of:
+
+- for management and application traffic
+- for application traffic only
+- unused
+
+Network port directly assigned to an application is without SystemAdapter.
 
-In addition, some network ports might be designated to application usage, or the underlying I/O bus controllers (e.g., PCI controllers) be designated for assignment to applications. That part of the configuration can also change.
+## Load spreading and failover
 
-Any network port configuration changes which might affect the connectivity to the controller requires care to avoid permanently losing connectivity to the controller and become unmanageable as a result. Thus it is required to perform testing as part of a configuration change and have a way to fall back to a working port configuration.
+Load spreading means that there are two or more similar uplink networks, for instance
+two Ethernet ports for redundancy, the EVE will send different requests over different
+connections in a round-robin fashion.
+
+Failover means that a device can have different types of uplink networks, for example
+Ethernet, LTE, and/or satellite connectivity, and switch from one to another to maintain
+controller connectivity. In such case, a cost (field ```cost``` of the
+[SystemAdapter message](https://github.com/lf-edge/eve-api/tree/main/proto/config/devmodel.proto))
+can be assigned to each uplink port so that e.g., LTE (the wwan0 port) is only used if EVE
+can not connect via any lower cost (e.g., one or more Ethernet) ports.
+
+The cost is a number between 0 and 255, with zero (the default) implying free,
+and less preferred ports being assigned a higher cost. Multiple ports can be assigned
+the same cost, in which case once failover has happened to cost N, then all uplink ports
+with cost N will be used for load spreading of the management traffic.
+
+Furthermore, one can set [network.download.max.cost](CONFIG-PROPERTIES.md) to a number N
+if it is acceptable for EVE to perform (potentially large) downloads of content and images
+when having failed over to uplink ports with cost N (or less).
+
+The cost is new and will replace the (now obsolete) `free`/`freeUplink` way to specify
```

Comment:
"new" and "will" might have been true in the past. 

Either we should completely remove the free/freeUplink from the documentation (which definitely makes sense if those old releases are no longer in use), or at least change the tense to say something like (Prior to the introduction of the cost integer there were free/freeUplink booleans which EVE still looks at to provide compatibility for old controllers.)